test/unit/tool_util/test_tool_linters.py

```diff
@@ -410,7 +410,7 @@
             <validator type="in_range">TEXT</validator>
             <validator type="regex" filename="blah"/>
             <validator type="expression"/>
-            <validator type="expression">[</validator>
+            <validator type="regex">[</validator>
```

Comment:
Maybe just add a validator here?

Reply:
Do you mean just calling `eval` with the expression text?

Reply:
I see. The test was just wrong. My bad.

But we could really call something like `compile(source, filename)` or `ast.parse(expr, filename='<unknown>', mode='exec')` on the expression content.

Reply:
Ahh sure, if there are ways to check the "syntax" without really evaluating I guess we can add it.
I'll investigate :+1:

Reply:
I used `ast.parse` and added a couple of test lines. Let's see if this works for all expressions :crossed_fingers:

Reply:
From the test failures, it seems different versions of Python return slightly different error messages for invalid expressions. I'll see if I can loosen the asserts a bit.

Reply:
I have relaxed the LintMessage comparison to check if the string is contained in the actual message instead of checking for the exact thing, I don't like that much that is implemented in the `__eq__` function, but it is only used for testing and we should take into account that error messages tend to change or improve over time without needing to fail the test when they do.
If this is not what we want, I can try a different more ad-hoc approach.